include/malloc_quda.h

```diff
@@ -45,15 +45,22 @@ namespace quda {
    */
   void *device_malloc_(const char *func, const char *file, int line, size_t size);
   void *device_pinned_malloc_(const char *func, const char *file, int line, size_t size);
+  void *device_comms_pinned_malloc_(const char *func, const char *file, int line, size_t size);
   void *safe_malloc_(const char *func, const char *file, int line, size_t size);
   void *pinned_malloc_(const char *func, const char *file, int line, size_t size);
   void *mapped_malloc_(const char *func, const char *file, int line, size_t size);
   void *managed_malloc_(const char *func, const char *file, int line, size_t size);
   void device_free_(const char *func, const char *file, int line, void *ptr);
   void device_pinned_free_(const char *func, const char *file, int line, void *ptr);
+  void device_comms_pinned_free_(const char *func, const char *file, int line, void *ptr);
   void managed_free_(const char *func, const char *file, int line, void *ptr);
   void host_free_(const char *func, const char *file, int line, void *ptr);
-
+#ifdef NVSHMEM_COMMS
+  void *shmem_malloc_(const char *func, const char *file, int line, size_t size);
+  void *shmem_pinned_malloc_(const char *func, const char *file, int line, size_t size);
+  void shmem_free_(const char *func, const char *file, int line, void *ptr);
+  void shmem_pinned_free_(const char *func, const char *file, int line, void *ptr);
+#endif
```

Comment:
These don't look like they are used outside of malloc, with the exception of the `sync_arr` allocation.  Can that not just use the abstracted `device_comms_pinned_malloc`, and not expose these to the outside world?

Reply:
Good idea. I think so.